controllers/imagehub/api/v1/repository_types.go

```diff
@@ -103,6 +104,16 @@ func (r *Repository) getName() string {
 	return r.Name
 }
 
+func (r *Repository) genKeywordsLabels(img *Image) {
+	if r.Labels == nil {
+		r.Labels = make(map[string]string)
+	}
+	for _, keyword := range img.Spec.DetailInfo.Keywords {
```

Comment:
the name is only supported by: 
```
The name segment is required and must be 63 characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between. 
```
Direct use of `DetailInfo.Keywords` will fail duo to some wrong words?

Reply:
solved. use a map to check.